pennylane_lightning/core/src/simulators/lightning_gpu/gates/tests/Test_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -1108,11 +1108,48 @@ TEMPLATE_TEST_CASE("StateVectorCudaManaged::SetStateVector",
             "the host") {
         auto init_state =
             createRandomStateVectorData<PrecisionT>(re, num_qubits);
-        auto expected_state = init_state;
+
+        StateVectorCudaManaged<TestType> sv{num_qubits};
+
+        std::vector<std::complex<PrecisionT>> values(init_state.begin(),
+                                                     init_state.end());
+
+        sv.setStateVector(values.data(), values.size(),
+                          std::vector<std::size_t>{0, 1, 2});
+        CHECK(init_state == Pennylane::Util::approx(sv.getDataVector()));
+    }
+
+    SECTION("Set state vector with values and their corresponding indices on "
+            "the host for a subset of wires right significant") {
+        auto init_state =
+            createRandomStateVectorData<PrecisionT>(re, num_qubits - 1);
+
+        std::vector<std::complex<PrecisionT>> expected_state(
+            Pennylane::Util::exp2(num_qubits), {0, 0});
+
+        std::copy(init_state.begin(), init_state.end(), expected_state.begin());
+
+        StateVectorCudaManaged<TestType> sv{num_qubits};
+
+        std::vector<std::complex<PrecisionT>> values(init_state.begin(),
+                                                     init_state.end());
+
+        sv.setStateVector(values.data(), values.size(),
+                          std::vector<std::size_t>{1, 2});
+        CHECK(expected_state == Pennylane::Util::approx(sv.getDataVector()));
+    }
+
+    SECTION("Set state vector with values and their corresponding indices on "
+            "the host for a subset of wires left significant") {
+        auto init_state =
+            createRandomStateVectorData<PrecisionT>(re, num_qubits - 1);
+
+        std::vector<std::complex<PrecisionT>> expected_state(
+            Pennylane::Util::exp2(num_qubits), {0, 0});
 
         for (std::size_t i = 0; i < Pennylane::Util::exp2(num_qubits - 1);
              i++) {
-            std::swap(expected_state[i * 2], expected_state[i * 2 + 1]);
+            expected_state[i * 2] = init_state[i];
```

Comment:
Sorry, but what is happening here?
```
            expected_state[i * 2] = init_state[i];
```
Why the expected state is defined like that?

Reply:
When the input state has left significant target wires (starting at zero and contiguous), the data is distributed along the base state vector with a stride. If the input vector has n-1 wires target, then the stride is 2. In general, the stride = 2 ** (n_qubits - n_targuet_wries)

Reply:
Thank you for your kind explanation. Suggestion:
```suggestion
            # Distributing along the base vector with a stride.  stride is 2 ** (n_qubits - n_target_wires)
            expected_state[i * 2] = init_state[i];
```

Reply:
You can move it to before the start of the for loop.